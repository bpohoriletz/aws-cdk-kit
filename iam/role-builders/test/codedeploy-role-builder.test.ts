import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import CodedeployRoleBuilder from '../codedeploy-role-builder';

describe('new CodedeployRoleBuilder()', () => {
  let instance: CodedeployRoleBuilder;
  let stack: cdk.Stack;

  beforeEach(() => {
    stack = new cdk.Stack();
    instance = new CodedeployRoleBuilder(stack, 'CodeDeployRole');
  });

  test('construct role for CodeDeploy', () => {
    instance.setManagedPolicies().setInlinePolicies(['*']);

    expect(Template.fromStack(stack)).toMatchSnapshot();
  });
});
