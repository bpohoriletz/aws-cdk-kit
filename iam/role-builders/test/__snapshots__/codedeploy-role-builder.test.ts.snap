// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new CodedeployRoleBuilder() construct role for CodeDeploy 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "CodeDeployRole12BEECBE": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "codedeploy.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSCodeDeployRole",
              ],
            ],
          },
        ],
        "RoleName": "CodeDeployRole",
      },
      "Type": "AWS::IAM::Role",
    },
    "allowEc2UploadLogsToCodedeployRegionalBucketF9632D95": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:ListBucket",
                "s3:ListBucketVersions",
                "s3:GetObject",
                "s3:GetObjectVersion",
                "s3:PutObject",
              ],
              "Effect": "Allow",
              "Resource": [
                "*",
                "*/resources/*",
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "allowEc2UploadLogsToCodedeployRegionalBucketF9632D95",
        "Roles": [
          {
            "Ref": "CodeDeployRole12BEECBE",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
